### PR TITLE
Учебный проект «Холст»

### PR DESCRIPTION
**Pull request by a201150209**
---
:mortar_board: [Учебный проект «Холст»](https://up.htmlacademy.ru/javascript/17/user/83947/tasks/3)

:boom: https://htmlacademy-javascript.github.io/83947-code-and-magick-17/2/

### CODE REVIEW

File: js/stat.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  };
+
+  var cloudColor = 'rgba(255, 255, 255, 1)';
+  var cloudShadowColor = 'rgba(0, 0, 0, 0.7)';
+  var mainColor = 'rgba(0, 0, 0, 1)';
+  var histogramHeight = 150;
+  var histogramColumnWidth = 40;
+  var histogramColumnPadding = 50;
+  var histogramUserColor = 'rgba(255, 0, 0, 1)';
+  var histogramUserName = 'Вы';
```
Comment (rkhaslarov) [inline js/stat.js]: please, refer to the guide and take a look how you should name the constants =)

File: js/stat.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    ctx.closePath();
+    ctx.fillStyle = color;
+    ctx.fill();
+  };
+
+  paintObject(cloudShadowCoodinates, );
+  paintObject(cloudCoordinates, cloudColor);
+
+  ctx.fillStyle = mainColor;
```
Comment (rkhaslarov) [inline js/stat.js]: the code should be covered with function and shouldn't be executed in global scope

File: js/stat.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+
+  var maxTime = times[0];
+  for (var i = 0; i < times.length; i++) {
+    times[i] = Math.round(times[i]);
+    if (times[i] > maxTime) {
+      maxTime = times[i];
+      var maxTimeElementIndex = i;
+    };
+  };
```
Comment (rkhaslarov) [inline js/stat.js]: each part of the code should be splitted into the separate function

File: js/stat.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+  for (var i = 0; i < names.length; i++) {
+    if (names[i] === histogramUserName) {
+      ctx.fillStyle = histogramUserColor;
+    } else {
+      var histogramOtherUsersColor = 'rgba(0, 0, 255, ' + Math.random(0.5, 1) + ')';
+      ctx.fillStyle = histogramOtherUsersColor;
+    };
```
Comment (rkhaslarov) [inline js/stat.js]: if (names[i] === histogramUserName) {
      ctx.fillStyle = histogramUserColor;
    } else {
      var histogramOtherUsersColor = 'rgba(0, 0, 255, ' + Math.random(0.5, 1) + ')';
      ctx.fillStyle = histogramOtherUsersColor;
    };

function?

File: js/stat.js (hunk 1 above)
Comment (a201150209) [reply js/stat.js]: Done

File: js/stat.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    paintObject(cloudShadow, cloudShadow.backgroundColor);
+    paintObject(cloud, cloud.backgroundColor);
+
+    ctx.fillStyle = cloud.mainColor;
+    ctx.font = '16px PT Mono';
+    ctx.fillText('Ура вы победили!', cloud.x + 60, cloud.y + 20);
+    ctx.fillText('Список результатов:', cloud.x + 60, cloud.y + 40);
+  })();
+
+  (function () {
```
Comment (rkhaslarov) [inline js/stat.js]: it's not needed, please remove it in scope of the next hw

**Pull request merged**